File: lib/github_media.rb
require 'uri'

class GithubMedia
  BASE_URL = 'http://github.com'

  # Wrapper for hosting files on Github.
  #
  #   options = {
  #     :user => 'Username',
  #     :repo => 'repo-name',
  #     :branch => 'master',
  #   }
  #   media = GithubMedia.new(options)
  def initialize(options = {})
    @options = options
    @base_path = [
      BASE_URL,
      @options[:user],
      @options[:repo],
      'raw',
      @options[:branch]
    ].join('/')
  end

  def link_to(filename)
    [@base_path, filename].join('/')
  end

end # GithubMedia

File: lib/python/github_media.py
from urlparse import urljoin

class GithubMedia(object):
  # Wrapper for hosting files on Github.
  #
  #   options = {
  #     'user': 'Username',
  #     'repo': 'repo-name',
  #     'branch': 'master',
  #   }
  #   media = GithubMedia(options)
  def __init__(self, options = {}):
    self.BASE_URL = 'http://github.com'
    self.options   = options
  
  def create_base_path(self):
    return urljoin(self.BASE_URL,
                   "/".join([self.options['user'],
                   self.options['repo'],
                   'raw',
                   self.options['branch'],]))
  base_path = property(create_base_path)

  def link_to(self, filename):
    return "/".join([self.base_path, filename])

